client/my-sites/stats/hooks/use-available-upgrade-tiers.tsx

```diff
@@ -72,10 +72,18 @@ function filterLowerTiers(
 	} else {
 		// Filter out tiers that have been purchased or lower than the current usage.
 		tiers = availableTiers.filter( ( availableTier ) => {
+			// `usageData?.current_tier?.limit` is used here to filter out the purchased tier without considering bundled views from Complete plans.
 			return (
 				!! availableTier.transform_quantity_divide_by ||
 				( availableTier?.views as number ) >
-					Math.max( usageData?.views_limit, usageData?.billableMonthlyViews, 0 )
+					Math.max(
+						usageData?.current_tier?.limit ?? 0,
+						// Offset the billable monthly views by the views already included in plans, e.g. COmplete.
+						usageData?.billableMonthlyViews -
```

Comment:
The bundled views are offset from here when calculating tiers for upgrades. How does this look to you @dognose24 ? Thanks

Reply:
It works well. However, could we refactor with variables to indicate the meaning of each property here?

- `usageData?.billableMonthlyViews`: Billable views count
- `usageData.views_limit`: Total views limit including the Complete plan
- `usageData.current_tier?.limit`: Current tier views limit of Stats commercial product

On the other hand, how about calculating the **_final_** `billableMonthlyViews` from the API end as you do here? 🤔

Reply:
`billableMonthlyViews` is calculated from the frontend at the moment. However that's a good idea to calculated everything from the backend, which would help with the bundle size.

> usageData?.billableMonthlyViews: Billable views count
usageData.views_limit: Total views limit including the Complete plan
usageData.current_tier?.limit: Current tier views limit of Stats commercial product

I'll add comment in this case as I feel like three more variables are a little too much here. Addressed 802dcefabd9